### PR TITLE
Add Befunge samples and .bf heuristics

### PR DESCRIPTION
**Pull request by smola**
## Description
Added heuristics for `.bf` extension.

Added the `.bf` extension to Befunge.

Added the following Befunge-93 samples:

* aturley.bf by Andrew Turley (CC0 license), fetched from https://github.com/catseye/Befunge-93/blob/master/eg/aturley.bf
* beer7.bf by Wim Rijnders (CC0 license), fetched from https://github.com/catseye/Befunge-93/blob/master/eg/beer7.bf
* befunge3.bf by Ben Olmstead (CC0 license), fetched from https://github.com/catseye/Befunge-93/blob/master/eg/befunge3.bf
* cascade.bf by Chris Pressey (CC0 license), fetched from https://github.com/catseye/Befunge-93/blob/master/eg/cascade.bf

And Befunge-98:

* mycorand.bf by Matti Niemenmaa (MIT license), fetched from https://github.com/Deewiant/Mycology/blob/master/mycorand.bf

For future reference: I have not included the (rarer) extensions `.b93` and `.b98`, but there are samples for it under liberal licenses here:

* mycoterm.b98 by Matti Niemenmaa (BSD license), fetched from https://github.com/Deewiant/Mycology/blob/master/mycoterm.b98
* Euler_Problem-001.b93 by Mike Schwörer (MIT License), fetched from https://github.com/Mikescher/Project-Euler_Befunge/blob/master/processed/Euler_Problem-001.b93

## Checklist:
- [x] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR (see above)
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by smola** [comment]
Although `.bf` seems to be the most common extension for Befunge, I can't tell if it meets the popularity threshold. I we're not including it, I'll create a separate PR for the `.bf` heuristics only.

File: lib/linguist/heuristics.yml [hunk 1]
```diff
@@ -88,6 +88,16 @@ disambiguations:
     pattern: '(<^\s*; |End Function)'
   - language: BitBake
     pattern: '^\s*(# |include|require)\b'
+- extensions: ['.bf']
+  rules:
+  - language: Beef
+    pattern: '^using\s+(System|Beefy)(\.(.*))?;\s*$'
```
Comment (Nixinova) [inline lib/linguist/heuristics.yml:94]: Wouldn't just `^using\s+(System|Beefy)` work fine? Also should there be a `\s*` before `using`?

Comment (Alhadis) [inline lib/linguist/heuristics.yml:94]: For performance's sake, I recommend disabling multiline mode so `(.*)` won't scan any further than 1 line:

```suggestion
    pattern: '(?-m)^\s*using\s+(System|Beefy)(\.(.*))?;\s*$'
```

File: lib/linguist/heuristics.yml [hunk 2]
```diff
@@ -88,6 +88,16 @@ disambiguations:
...
+  - language: HyPhy
+    pattern:
+    - '^\s*#include\s+".*";\s*$'
```
Comment (Alhadis) [inline lib/linguist/heuristics.yml:97]: Ditto.
```suggestion
    - '(?-m)^\s*#include\s+".*";\s*$'
```

**Review by Alhadis — CHANGES_REQUESTED** [review]

**Comment by lildude** [comment]
Looks like the classifier cross-validation is struggling with one of the Befunge samples causing the test to fail:

```
Befunge/mycorand.bf BAD (Brainfuck)
```

**Review by Alhadis — APPROVED** [review]

**Review by lildude — APPROVED** [review]

**Pull request merged**